server/ServerFSM.cpp

```diff
@@ -2324,8 +2324,8 @@ sc::result ProcessingTurn::react(const ProcessTurn& u) {
         }
     }
 
-    if (server.IsHostless() && GetOptionsDB().Get<bool>("save.server.hostless")) {
-        boost::filesystem::path autosave_dir_path = GetServerSaveDir() / "auto";
```

Comment:
why do you say the auto save subdirectory is unused...?

Reply:
It looks like it checks that the save path is in the save directory, not that it is the save directory.

Reply:
It also checks if `auto` directory exists but don't create it.

Reply:
ok... so create it then? autosaves are supposed to be segregated into the auto directory

Reply:
Ok, I'll look into it.

Reply:
Looks like `boost::filesystem::canonical` in `IsInDir` requires directories to exist. There is a simple solution with `boost::filesystem::weakly_canonical` but it was introduced in boost 1.60

Reply:
> Looks like boost::filesystem::canonical in IsInDir requires directories to exist.

What is the importance / consequence of this?

Reply:
> What is the importance / consequence of this?
It throws error if path not exists: http://www.boost.org/doc/libs/1_58_0/libs/filesystem/doc/reference.html#canonical
Yet I get a little  workaround around it.